### DISCUSSION TITLE
How to use API on Hiddify project | راهنمای چگونگی استفاده از API در پروژه هیدیفای

### DISCUSSION THREAD

**Discussion by lymanjre** [hiddify/Hiddify-Manager#3209] [Q&A سوالات رایج] [answered]
If you want to use API on Hiddify project, how to use it?
اگر بخواهید با استفاده از API در پروژه هیدیفای همکاری کنید، چگونه از آن استفاده کنید؟

**Comment by lymanjre** [comment, marked as answer]
در اینجا راهنمای استفاده از API در هیدیفای ارائه می‌‌گردد که در دو بخش کار با یوزرهای معمولی و کار با یوزرهای ادمین می‌آید.

* برای دریافت لیست همه یوزرها
 ``` 
GET https://site.com/proxy_path/admin_key/api/v1/user/
``` 

* برای گرفتن یک یوزر خاص
``` 
 GET https://site.com/proxy_path/admin_key/api/v1/user/?uuid=6ebd2ea8-4d41-48b7-8fc2-7d6570
``` 

* برای ایجاد یا آپدیت یوزر

``` 
POST https://site.com/proxy_path/admin_key/api/v1/user/
        {
            "uuid": "6ebd2ea8-4d41-48b7-8fc2-7d6570da30a9",
            "name": "Test",
            "added_by_uuid": "abb4bdb8-732c-4fa6-92e0-6b3fd4bb8450",
            "current_usage_GB": 0,
            "usage_limit_GB": 1,
            "package_days": 900,
            "start_date": null,
            "comment": null,
            "last_online": "1-01-01 00:00:00",
            "last_reset_time": "2023-05-21",
            "mode": "no_reset",
            "telegram_id": null
        }
``` 
> نکته: فیلد ed25519_private_key و ed25519_public_key اختیاری است و در صورتی که نباشد توسط سیستم تولید می‌شود.

* برای گرفتن همه یوزرهای ادمین
``` 
GET https://site.com/proxy_path/admin_key/api/v1/admin/
``` 
* برای گرفتن  یک یوزر ادمین
``` 
GET https://site.com/proxy_path/admin_key/api/v1/admin/?uuid=abb4bdb8-732c-4fa6-92e0-6b3fd4bb8450
``` 
* برای ایجاد یا آپدیت یوزر ادمین
``` 
POST https://site.com/proxy_path/admin_key/api/v1/admin/
        {
            "uuid": "0749f63d-16c1-4bdf-8d4f-180bcd4cdef6",
            "can_add_admin": false,
            "mode": "agent",
            "name": "Hiddify-Admin",
            "parent_admin_uuid": "abb4bdb8-732c-4fa6-92e0-6b3fd4bb8450",
            "comment": null,
            "telegram_id": null
        }
``` 
* فرمت url مربوط به یوزر
``` 
https://site.com/proxy_path/uuid/
``` 
* برای گرفتن لینک کوتاه صفحه‌ی کاربر
```
GET https://site.com/proxy_path/user_uuid/short/
```
* برای گرفتن اطلاعات کاربر
```
GET https://site.com/proxy_path/user_uuid/info/
```
* برای گرفتن لیست پروکسی های تلگرام کاربر
```
GET https://site.com/proxy_path/user_uuid/mtproxies/
```
* برای گرفتن لینک تمام کانفیگ های کاربر
```
GET https://site.com/proxy_path/user_uuid/all-configs/
```

**Reply by xmohammad1** [reply]
امکانش هست api حذف و فعال و غیرفعال کردن کاربر رو هم اضافه کنید ، قابلیت استعلام فعال یا غیر فعال بودن کاربر هم باشه

**Reply by lankze** [reply]
> در اینجا راهنمای استفاده از API در هیدیفای ارائه می‌‌گردد که در دو بخش کار با یوزرهای معمولی و کار با یوزرهای ادمین می‌آید.
> 
> * برای دریافت لیست همه یوزرها
> 
> ```
> GET https://site.com/proxy_path/admin_key/api/v1/user/
> ```
> 
> * برای گرفتن یک یوزر خاص
> 
> ```
>  GET https://site.com/proxy_path/admin_key/api/v1/user/?uuid=6ebd2ea8-4d41-48b7-8fc2-7d6570
> ```
> 
> * برای ایجاد یا آپدیت یوزر
> 
> ```
> POST https://site.com/proxy_path/admin_key/api/v1/user/
>         {
>             "uuid": "6ebd2ea8-4d41-48b7-8fc2-7d6570da30a9",
>             "name": "Test",
>             "added_by_uuid": "abb4bdb8-732c-4fa6-92e0-6b3fd4bb8450",
>             "current_usage_GB": 0,
>             "usage_limit_GB": 1,
>             "package_days": 900,
>             "start_date": null,
>             "comment": null,
>             "last_online": "1-01-01 00:00:00",
>             "last_reset_time": "2023-05-21",
>             "mode": "no_reset",
>             "telegram_id": null
>         }
> ```
> 
> > نکته: فیلد ed25519_private_key و ed25519_public_key اختیاری است و در صورتی که نباشد توسط سیستم تولید می‌شود.
> 
> * برای گرفتن همه یوزرهای ادمین
> 
> ```
> GET https://site.com/proxy_path/admin_key/api/v1/admin/
> ```
> 
> * برای گرفتن  یک یوزر ادمین
> 
> ```
> GET https://site.com/proxy_path/admin_key/api/v1/admin/?uuid=abb4bdb8-732c-4fa6-92e0-6b3fd4bb8450
> ```
> 
> * برای ایجاد یا آپدیت یوزر ادمین
> 
> ```
> POST https://site.com/proxy_path/admin_key/api/v1/admin/
>         {
>             "uuid": "0749f63d-16c1-4bdf-8d4f-180bcd4cdef6",
>             "can_add_admin": false,
>             "mode": "agent",
>             "name": "Hiddify-Admin",
>             "parent_admin_uuid": "abb4bdb8-732c-4fa6-92e0-6b3fd4bb8450",
>             "comment": null,
>             "telegram_id": null
>         }
> ```
> 
> * فرمت url مربوط به یوزر
> 
> ```
> https://site.com/proxy_path/uuid/
> ```
> 
> * برای گرفتن لینک کوتاه صفحه‌ی کاربر
> 
> ```
> GET https://site.com/proxy_path/user_uuid/short/
> ```
> 
> * برای گرفتن اطلاعات کاربر
> 
> ```
> GET https://site.com/proxy_path/user_uuid/info/
> ```
> 
> * برای گرفتن لیست پروکسی های تلگرام کاربر
> 
> ```
> GET https://site.com/proxy_path/user_uuid/mtproxies/
> ```
> 
> * برای گرفتن لینک تمام کانفیگ های کاربر
> 
> ```
> GET https://site.com/proxy_path/user_uuid/all-configs/
> ```

is this still working for the latest version?

**Comment by rezahajrahimi** [comment]
آدرس دریافت مشخصات برای گرفتن یک یوزر خاص را وارد می کنم اما باز لیست کاربران برگشت داده می شه

**Comment by elia07** [comment]
سلام و تشکر بابت زحماتی که تیمتون داره میکشه.
من Api ، User/Get ام کار میکنه اما وقتی میخوام با Api یه یوزر جدید بسازم اررور 502 bad gateway میده
متن زیر کپی شده از postman هست که اطلاعات مهم توش پاک شده
ممنون میشم راهنمایی کنید چه پارامتری رو زیاد و کم گذاشتم که این اررور رو می گیرم 

wget --no-check-certificate --quiet \
  --method POST \
  --timeout=0 \
  --header 'Content-Type: application/json' \
  --body-data '{
"uuid": "6ebd2ea8-4d41-48b7-8fc2-7d6570da30a9",
"name": "Testdefsdfsdfsdf",
"added_by_uuid": "ADMINAPIKEY",
"current_usage_GB": 0,
"usage_limit_GB": 1,
"package_days": 900,
"start_date": null,
"comment": null,
"last_online": "1-01-01 00:00:00",
"last_reset_time": "2023-05-21",
"mode": "no_reset",
"telegram_id": null
}' \
   'https://mydomain.com/Proxy_PAth/ADMINAPIKEY/api/v1/user/'

**Reply by elia07** [reply]
الان که دارم میبینم یوزر رو ایجاد میکنه اما 502 bad gateway میده . این احتمالا یه باگه.

**Comment by husseinttimani** [comment]
میشه راهنمایی بفرمایید چرا متد پست برای من کار نمیکنه و مدام این پیام میگیرم؟
{"message": "The browser (or proxy) sent a request that this server could not understand."}

**Comment by elia07** [comment]
سلام ،
ببخشید که دیر جواب دادم زیر فشار پروژه بودم
والا این آی پی آی کلا مشکل داشت و من چون خودم دارم از هدیفای استفااده می
کنم تصمیم گرفتم ایرادشو رفع کنم و پس از رفع شدنش هنوز توی پروژه خودم
اینتگریتش کردم که ظرف چند روز آینده قصدشو دارم
این مشکل شما رم بررسی می کنم بهتون می گم

**Reply by v4mpir3** [reply]
ممنونم
